packages/components/src/templates/next/components/complex/Hero/HeroLargeImage/ImageContainer.tsx

```diff
@@ -0,0 +1,66 @@
+"use client"
+
+import { useEffect, useRef, useState } from "react"
+
+import type { ImageClientProps } from "~/interfaces"
+import { ImageClient } from "../../Image"
+import { ScrollForMoreButton } from "./ScrollForMoreButton"
+
+interface ImageContainerProps {
+  imageSrc: ImageClientProps["src"]
+  imageAlt: ImageClientProps["alt"]
+}
+
+// Arbitrary threshold to prevent the button from showing too early
+const SCROLL_THRESHOLD = 120
+
+export const ImageContainer = ({ imageSrc, imageAlt }: ImageContainerProps) => {
+  const imageRef = useRef<HTMLImageElement>(null)
+  const [isFixed, setIsFixed] = useState(false)
+  const [shouldShowButton, setShouldShowButton] = useState(false)
+
+  const handleScroll = () => {
+    if (!imageRef.current) return
+
+    const imageRect = imageRef.current.getBoundingClientRect()
+
+    if (imageRect.top + SCROLL_THRESHOLD >= window.innerHeight) {
+      setShouldShowButton(false)
+      return
+    } else {
+      setShouldShowButton(true)
+    }
```

Comment:
[minor nit, feel free to ignore] can just directly `setShouldShowButton(condition)`